.pre-commit-config.yaml

```diff
@@ -38,6 +38,25 @@ repos:
   hooks:
     - id: black
       language_version: python3 # Should be a command that runs python3.6+
+- repo: https://github.com/jazzband/pip-tools
```

Comment:
We need to be careful with this. Having pip-tools in pre-commit is tricky because it resolves the dep tree for the current runtime. For the devs on macOS or Windows, or under a different CPython version, it will produce updates which is not what we want. In some other projects we have `pip-tools` set up as a local hook because of this, with `default_language_version:` for python set to 3.9. So I would exclude this part of the patch from this PR for now in order to see what would the consequences be separately.